tests/matrix_free/matrix_vector_hessians_faces.cc

```diff
@@ -0,0 +1,388 @@
+                      if (evaluation_flags & EvaluationFlags::values)
+                        values += solution_values_p(i) *
+                                  fe_face_values_p.shape_value(i, q);
+                    }
+                  solution_hessians[q]  = hessians * fe_face_values_p.JxW(q);
+                  solution_gradients[q] = gradients * fe_face_values_p.JxW(q);
+                  solution_values[q]    = values * fe_face_values_p.JxW(q);
+                }
+              for (unsigned int i = 0; i < fe.dofs_per_cell; ++i)
+                {
+                  double sum_hessians  = 0.;
+                  double sum_gradients = 0.;
+                  double sum_values    = 0.;
+                  for (unsigned int q = 0; q < quad.size(); ++q)
+                    {
+                      if (evaluation_flags & EvaluationFlags::hessians)
+                        sum_hessians += double_contract<0, 0, 1, 1>(
+                          solution_hessians[q],
+                          fe_face_values_p.shape_hessian(i, q));
+                      if (evaluation_flags & EvaluationFlags::gradients)
+                        sum_gradients += solution_gradients[q] *
+                                         fe_face_values_p.shape_grad(i, q);
+                      if (evaluation_flags & EvaluationFlags::values)
+                        sum_values += solution_values[q] *
+                                      fe_face_values_p.shape_value(i, q);
+                    }
+                  solution_values_p(i) =
+                    sum_hessians + sum_gradients + sum_values;
+                }
+              constraints.distribute_local_to_global(solution_values_p,
+                                                     dof_indices_p,
+                                                     dst2);
+            }
+        }
+    },
+    [&](
+      const auto &matrix_free, auto &dst, const auto &src, const auto &range) {
+      (void)matrix_free;
+      (void)dst;
+      (void)src;
+      (void)range;
+    },
+    dst,
+    src,
+    true);
+
+  const unsigned int end_of_print_dst =
+    dof_handler.n_dofs() > 9 ? 9 : dof_handler.n_dofs();
+
+  deallog << "dst FEE: ";
+  for (unsigned int i = 0; i < end_of_print_dst; ++i)
+    deallog << dst[i] << " ";
+  deallog << std::endl;
+
+  deallog << "dst FEV: ";
+  for (unsigned int i = 0; i < end_of_print_dst; ++i)
+    deallog << dst2[i] << " ";
+  deallog << std::endl;
+
+  // compare solutions of matrix vector product
+  {
+    dst2 -= dst;
+
+    double error = 0.;
+    if (dst.l2_norm() > 0)
+      error = dst2.l2_norm() / dst.l2_norm();
+    else
+      error = dst2.l2_norm();
+
+    if (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0)
+      deallog << "FEValues verification: " << error << std::endl << std::endl;
+  }
+}
+
+int
+main(int argc, char **argv)
+{
+  dealii::Utilities::MPI::MPI_InitFinalize mpi(argc, argv, 1);
+
+  initlog();
```

Comment:
```suggestion
  initlog();
  deallog << std::setprecision(10);
```
I would add this to be on the safe side for the large numbers appearing here and the relative accuracy of 1e-8 applied by numdiff.

Reply:
Done!